### PR TITLE
Refactor server to use the Echo framework

### PR DESCRIPTION
**Pull request by miketonks-form3**
Work towards of investment time project: https://github.com/form3tech/investment-time-project/issues/62

The Echo framework provides simple reusable code to make the setup, routing and handler code cleaner and more readable.

Echo is a good choice as it is fast, mature and stable.  It is very similar to gin, but has a newer code base, is a more active project, has a few improvements particularly around routing and error handling.

### CODE REVIEW

File: Makefile
Comment (sata-form3) [inline Makefile:44]: While we're changing this, can we do `-sSLf`

```
       -s, --silent
      -S, --show-error
              When used with -s, --silent, it makes curl show an error message if it fails.
       -L, --location
       -f, --fail
```

perhaps all the options directly after curl, rather it being in the middle?  `-o` could remain where it is I guess.

File: internal/app/pactproxy/proxy.go
Comment (miketonks-form3) [inline internal/app/pactproxy/proxy.go:83]: Bind (unmarshal) into structs is another thing I like - just a helper function but works well

File: internal/app/pactproxy/proxy.go
Comment (miketonks-form3) [inline internal/app/pactproxy/proxy.go:119]: Getting rid of these `if req.Method ...` is a big win for me.  I know it's only a small thing but it seems clumsy.

File: internal/app/pactproxy/proxy.go
Comment (miketonks-form3) [inline internal/app/pactproxy/proxy.go:169]: Looks like we don't check the http method at all in this handler?  Easy to do, since it's not checked by the router

**Comment by miketonks-form3** [comment]
I'd agree it's mostly just boiler plate, and how we configure the server etc is no big deal.

What is more important in terms of adding new features is making the handlers easy to work with.  I've added a few comments inline to show where I think some of the wins are.

If it would make it easier to get approval we could switch to gin instead of echo.  TBH I see echo as a natural upgrade from gin, they're very similar, for me it just has a few useful improvements.

**Comment by sata-form3** [comment]
You convinced me, I think we would be able to migrate to what ever comes next after Echo as well as the interfaces would be similar. Could we do the curl fix, and I then can approve if you don't mind

File: Makefile
Comment (miketonks-form3) [reply Makefile]: Done :+1:

**Review by sata-form3 — APPROVED** [review]

**Pull request merged**
